be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -47,31 +55,72 @@ class SinkBuffer {
     }
 
     ~SinkBuffer() {
-        _pending_chunks.shutdown();
+        _is_finished = true;
```

Comment:
~SinkBuffer is invoked when QueryContext is destructed, so SinkBuffer::process exits when here _is_finished = true,  if thread wait at location  _cache_empty_cv.wait(l), _cache_empty_cv's destruction is not sound.

Reply:
fixed